src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -62,6 +64,242 @@ static ShardPlacement * TupleToShardPlacement(TupleDesc tupleDesc,
 /* exports for SQL callable functions */
 PG_FUNCTION_INFO_V1(master_stage_shard_row);
 PG_FUNCTION_INFO_V1(master_stage_shard_placement_row);
+PG_FUNCTION_INFO_V1(citus_table_size);
+PG_FUNCTION_INFO_V1(citus_total_relation_size);
+PG_FUNCTION_INFO_V1(citus_relation_size);
+
+/*
+ * GetShardIntervalsOnNode takes WorkerNode then compare it with each placement of table.
+ * It returns intervals of table on that node as a list of intervals.
+ */
+List *
+GetShardIntervalsOnNode(WorkerNode *workerNode, Oid relationId)
+{
+	DistTableCacheEntry *distTableCacheEntry = DistributedTableCacheEntry(relationId);
+	char *workerNodeName = workerNode->workerName;
+	uint32 workerNodePort = workerNode->workerPort;
+	List *shardIntervalList = NIL;
+	int shardIndex = 0;
+	int shardIntervalArrayLength = distTableCacheEntry->shardIntervalArrayLength;
+
+	for (shardIndex = 0; shardIndex < shardIntervalArrayLength; shardIndex++)
+	{
+		ShardPlacement *placementArray =
+			distTableCacheEntry->arrayOfPlacementArrays[shardIndex];
+		int numberOfPlacements =
+			distTableCacheEntry->arrayOfPlacementArrayLengths[shardIndex];
+		int i = 0;
```

Comment:
Let's use placementIndex instead of i as a variable name. In whole codebase there are only 5 places where we used variable name "i". 4 of them are used mistakenly :smile: